packages/firestore/lite/src/api/reference.ts

```diff
@@ -49,3 +55,222 @@ export class DocumentReference<T = firestore.DocumentData>
     return new DocumentReference<U>(this.firestore, this._key, converter);
   }
 }
+
+export class Query<T = firestore.DocumentData> implements firestore.Query<T> {
+  constructor(
+    readonly firestore: Firestore,
+    readonly _query: InternalQuery,
+    readonly _converter?: FirestoreDataConverter<T>
+  ) {}
+
+  where(
+    fieldPath: string | firestore.FieldPath,
+    opStr: firestore.WhereFilterOp,
+    value: unknown
+  ): firestore.Query<T> {
+    // TODO(firestorelite): Implement
+    throw new Error('Not implemented');
+  }
+
+  orderBy(
+    fieldPath: string | firestore.FieldPath,
+    directionStr?: firestore.OrderByDirection
+  ): firestore.Query<T> {
+    // TODO(firestorelite): Implement
+    throw new Error('Not implemented');
+  }
+
+  limit(limit: number): firestore.Query<T> {
+    // TODO(firestorelite): Implement
+    throw new Error('Not implemented');
+  }
+
+  limitToLast(limit: number): firestore.Query<T> {
+    // TODO(firestorelite): Implement
+    throw new Error('Not implemented');
+  }
+
+  startAfter(
+    docOrField: unknown | firestore.DocumentSnapshot<unknown>,
+    ...fields: unknown[]
+  ): firestore.Query<T> {
+    // TODO(firestorelite): Implement
+    throw new Error('Not implemented');
+  }
+
+  startAt(
+    docOrField: unknown | firestore.DocumentSnapshot<unknown>,
+    ...fields: unknown[]
+  ): firestore.Query<T> {
+    // TODO(firestorelite): Implement
+    throw new Error('Not implemented');
+  }
+
+  endAt(
+    docOrField: unknown | firestore.DocumentSnapshot<unknown>,
+    ...fields: unknown[]
+  ): firestore.Query<T> {
+    // TODO(firestorelite): Implement
+    throw new Error('Not implemented');
+  }
+
+  endBefore(
+    docOrField: unknown | firestore.DocumentSnapshot<unknown>,
+    ...fields: unknown[]
+  ): firestore.Query<T> {
+    // TODO(firestorelite): Implement
+    throw new Error('Not implemented');
+  }
+
+  withConverter<U>(
+    converter: firestore.FirestoreDataConverter<U>
+  ): firestore.Query<U> {
+    return new Query<U>(this.firestore, this._query, converter);
+  }
+}
+
+export class CollectionReference<T = firestore.DocumentData> extends Query<T>
```

Comment:
Stripped down version of https://osscs.corp.google.com/firebase-sdk/firebase-js-sdk/+/master:packages/firestore/src/api/database.ts;l=2274